Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/CheckInsTransformer.kt

```diff
@@ -61,16 +62,38 @@ class CheckInsTransformer @Inject constructor(
                     checkInStart = derivedTimes.startTimeSeconds.secondsToInstant(),
                     checkInEnd = derivedTimes.endTimeSeconds.secondsToInstant()
                 )
-                derivedCheckIn.splitByMidnightUTC().map { checkIn ->
+                derivedCheckIn.splitByMidnightUTC().mapNotNull { checkIn ->
                     checkIn.toOuterCheckIn(transmissionVector)
                 }
             }
         }
     }
 
-    private fun CheckIn.toOuterCheckIn(
+    private suspend fun CheckIn.toOuterCheckIn(
         transmissionVector: TransmissionRiskVector
-    ): CheckInOuterClass.CheckIn {
+    ): CheckInOuterClass.CheckIn? {
+        val trvMappings: List<TransmissionRiskValueMapping> =
+            appConfigProvider.getAppConfig()
+                .presenceTracing
+                .riskCalculationParameters
+                .transmissionRiskValueMapping
+
+        val transmissionRiskLevel = determineRiskTransmission(
+            timeStamper.nowUTC,
```

Comment:
Should we maybe get the timestamp once, like the config, and then use it for every calculation, so that it doesn't change during calculation?

Reply:
`timeStamper.nowUTC`  is used only here

Reply:
But "here" is in `toOuterCheckIn` which is called from `mapNotNull` which is called from `flatMap`, so this is called many times, and depending on how fast the whole operation is, will very likely deliver different time values.

Using a timestamp that changes during calculation on the same dataset might at be problematic? At best difficult to compare results, at worst unexpected results if this happens around 23:59:59?